pkg/apis/migration/v1alpha1/directimagemigration_types.go

```diff
@@ -89,6 +92,24 @@ func (r *DirectImageMigration) GetDestinationCluster(client k8sclient.Client) (*
 	return GetCluster(client, r.Spec.DestMigClusterRef)
 }
 
+// Get the MigMigration that owns this DirectImageMigration. If not owned, return nil.
+func (r *DirectImageMigration) GetMigrationForDIM(client k8sclient.Client) (*MigMigration, error) {
+	owner := &MigMigration{}
+	ownerRefs := r.GetOwnerReferences()
+	for _, ownerRef := range ownerRefs {
+		if ownerRef.Kind != "MigMigration" {
+			continue
+		}
+		ownerRef := types.NamespacedName{Name: ownerRef.Name, Namespace: r.Namespace}
+		err := client.Get(context.TODO(), ownerRef, owner)
+		if err != nil {
+			return nil, liberr.Wrap(err)
```

Comment:
What happens here on a 500 error from the API server? what happens if there is a 404? 

What I would be worried about is a slow cache update I think  right?

Reply:
I don't believe we've had any notable problems with cache being out of date for the local cluster. The outdated cache issue primarily showed up as an issue communicating with remote clusters. All of our CRs are located on the same cluster as us.

The strategy for creating these spans is best-effort: if we can find the associated resources then we'll create a span. Otherwise we ignore it and the migration proceeds without jaeger spans. The jaeger span is not necessary for migration success, just helps us understand the system.

https://github.com/konveyor/mig-controller/pull/967/files/bc6c07c64755c415f2b1c8ae09acb4caca4e6f74#diff-c1e9e79d4cf358ba29d2e6ec3d769c82cd837a9269aae54f489504c64dbb672bR53-R58